pkg/packageinstall/packageinstall.go

```diff
@@ -210,6 +210,16 @@ func (pi *PackageInstallCR) referencedPkgVersion() (datapkgingv1alpha1.Package,
 		}
 	}
 
+	// If constraint is a single specified version, then we
+	// do not want to force user to manually set prereleases={}
+	if len(semverConfig.Constraints) > 0 && semverConfig.Prereleases == nil {
+		// Will error if it's not a single version
+		singleVer, err := versions.NewSemver(semverConfig.Constraints)
```

Comment:
Are we ignoring the error because a following stage will error instead?

Reply:
constraints could be various things: single version or version ranges. if we fail to parse it as single version then it's either malformed version or it's not a single version. in either one of those cases we dont want to mangle what the user provided since we do not know what it is.

Reply:
oh nice, I didn't realise it could be a range. Makes sense!